WallRelatedPages::getMessagesRelatedArticleIds - one does not simply call wfGetDB() twice

macbre opened the pull request:
Prevent `PHP Notice: Object of class DatabaseMysqli could not be converted to int in /usr/wikia/slot1/13783/src/includes/db/LoadBalancer.php on line 491`

followed by

```
Unexpected non-MediaWiki exception encountered, of type "Error"
Error: Call to a member function setLBInfo() on null in /usr/wikia/slot1/13783/src/includes/db/LoadBalancer.php:745
```

Causes `ForumBoardTests` to fail on staging.

Force `wfGetDB`'s `$db`  argument to be int (PHP7 feature), to catch other cases where Database class instance is passed there - f193ce2

``` php
> wfGetDB(DB_SLAVE)
LoadBalancer::getReaderIndex: Using reader #1: slave.db-dev-db.service.consul...
Connecting to slave.db-dev-db.service.consul plpoznan...

> wfGetDB('foo')
Unexpected non-MediaWiki exception encountered, of type "TypeError"
TypeError: Argument 1 passed to wfGetDB() must be of the type integer, string given, called in maintenance/eval.php(88) : eval()'d code on line 1 and defined in includes/GlobalFunctions.php:3623
```

@mixth-sense / @sqreek / @TK-999

macbre committed:
WallRelatedPages::getMessagesRelatedArticleIds - one does not simply call wfGetDB() twice

Prevent PHP Notice: Object of class DatabaseMysqli could not be converted to int in /usr/wikia/slot1/13783/src/includes/db/LoadBalancer.php on line 491

followed by

Unexpected non-MediaWiki exception encountered, of type "Error"
Error: Call to a member function setLBInfo() on null in /usr/wikia/slot1/13783/src/includes/db/LoadBalancer.php:745

WikiaTech commented:
SonarQube analysis reported 3 issues
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 3 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [WallRelatedPages.class.php#L14](https://github.com/Wikia/app/blob/7d0af727f6c47c0f003d36047848da4d9790d661/extensions/wikia/Wall/WallRelatedPages.class.php#L14): Define a constant instead of duplicating this literal "wall_related_pages" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [WallRelatedPages.class.php#L54](https://github.com/Wikia/app/blob/7d0af727f6c47c0f003d36047848da4d9790d661/extensions/wikia/Wall/WallRelatedPages.class.php#L54): Define a constant instead of duplicating this literal "comment_id" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/Wall/WallRelatedPages.class.php:
```diff
@@ -115,20 +115,20 @@ function getMessageRelatetMessageIds( $messageId ) {
 	 */
 
-	function getMessagesRelatedArticleIds( $messageIds, $orderBy = 'order_index', $db = DB_SLAVE ) {
+	function getMessagesRelatedArticleIds( $messageIds, $orderBy = 'order_index', $dbType = DB_SLAVE ) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "order_index" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech replied:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "order_index" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

macbre committed:
wfGetDB: expect $db argument to be an integer

> wfGetDB(DB_SLAVE)
LoadBalancer::getReaderIndex: Using reader #1: slave.db-dev-db.service.consul...
Connecting to slave.db-dev-db.service.consul plpoznan...

> wfGetDB('foo')
Unexpected non-MediaWiki exception encountered, of type "TypeError"
TypeError: Argument 1 passed to wfGetDB() must be of the type integer, string given, called in maintenance/eval.php(88) : eval()'d code on line 1 and defined in includes/GlobalFunctions.php:3623

WikiaTech commented:
SonarQube analysis reported 93 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 40 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 27 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php): File "GlobalFunctions.php" has 4098 lines, which is greater than 1000 authorized. Split it into smaller files. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L66](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L66): Remove the unused function parameter "$var". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L70](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L70): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L72](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L72): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L140](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L140): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L196](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L196): Remove the unused function parameter "$array1". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L229](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L229): Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L396](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L396): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L507](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L507): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L647](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L647): This branch's code block is the same as the block for the branch on line 637. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)

WikiaTech commented:
SonarQube analysis reported 93 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 40 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 27 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php): File "GlobalFunctions.php" has 4098 lines, which is greater than 1000 authorized. Split it into smaller files. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L66](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L66): Remove the unused function parameter "$var". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L70](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L70): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L72](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L72): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L140](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L140): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L196](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L196): Remove the unused function parameter "$array1". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L229](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L229): Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L396](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L396): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L507](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L507): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L647](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L647): This branch's code block is the same as the block for the branch on line 637. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)

WikiaTech commented:
SonarQube analysis reported 93 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 40 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 27 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 26 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php): File "GlobalFunctions.php" has 4098 lines, which is greater than 1000 authorized. Split it into smaller files. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L66](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L66): Remove the unused function parameter "$var". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L70](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L70): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L72](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L72): Remove the unused function parameters "$var, $level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L140](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L140): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L196](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L196): Remove the unused function parameter "$array1". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L229](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L229): Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L396](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L396): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L507](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L507): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [GlobalFunctions.php#L647](https://github.com/Wikia/app/blob/f193ce230e04605d4cb7f99acd34ed27beedc5cc/includes/GlobalFunctions.php#L647): This branch's code block is the same as the block for the branch on line 637. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)

mixth-sense commented:
:+1: LGTM
